tests/conftest.py

```diff
@@ -226,7 +227,7 @@ def duthost(duthosts, request):
     @param request: request parameters for duthost test fixture
     '''
     dut_index = getattr(request.session, "dut_index", 0)
-    assert dut_index < len(duthosts), \
+    assert dut_index < len(duthosts.nodes), \
         "DUT index '{0}' is out of bound '{1}'".format(dut_index,
                                                        len(duthosts))
```

Comment:
Here `len(duthosts)` needs the implementation of `__len__` method on `DutHosts`.